client-hc/src/main/java/com/graphhopper/api/MatrixResponse.java

```diff
@@ -13,8 +13,12 @@
 
     private String debugInfo = "";
     private final List<Throwable> errors = new ArrayList<>(4);
+    private final List<PointPair> disconnectedPoints = new ArrayList<>(0);
+    // todonow: should we add hppc to classpath and use IntArrayList instead ?
```

Comment:
wasn't sure here

Reply:
Yeah those cases of small objects in a list are ultra memory waste but we should improve this later if this makes a problem for bigger matrices.

Reply:
we could even return an int[] ?